File: backend/src/main/java/io/chesslave/app/JsonEventConverter.java
package io.chesslave.app;

import com.fasterxml.jackson.databind.DeserializationFeature;
import com.fasterxml.jackson.databind.ObjectMapper;
import java.io.IOException;

public class JsonEventConverter implements Converter<Event> {

    private final ObjectMapper mapper;

    public JsonEventConverter() {
        this.mapper =   new ObjectMapper();
        this.mapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
    }

    @Override
    public String asString(Event event) throws IOException {
        return mapper.writeValueAsString(event);
    }

    @Override
    public Event fromString(String text) throws IOException {
        return mapper.readValue(text, Event.class);
    }
}

File: backend/src/main/java/io/chesslave/app/JsonEventConverter.kt
package io.chesslave.app

import com.fasterxml.jackson.databind.DeserializationFeature
import com.fasterxml.jackson.databind.ObjectMapper

class JsonEventConverter : Converter<Event> {

    private val mapper = ObjectMapper().disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)

    override fun asString(event: Event): String = mapper.writeValueAsString(event)

    override fun fromString(text: String): Event = mapper.readValue(text, Event::class.java)
}
